addons/sale_timesheet/models/account_move.py

```diff
@@ -11,30 +11,32 @@ class AccountMove(models.Model):
     _inherit = "account.move"
 
     timesheet_ids = fields.One2many('account.analytic.line', 'timesheet_invoice_id', string='Timesheets', readonly=True, copy=False)
-    timesheet_count = fields.Integer("Number of timesheets", compute='_compute_timesheet_count')
+    timesheet_count = fields.Integer("Number of timesheets", compute='_compute_timesheet_data', compute_sudo=True)
     timesheet_encode_uom_id = fields.Many2one('uom.uom', related='company_id.timesheet_encode_uom_id')
-    timesheet_total_duration = fields.Integer("Timesheet Total Duration", compute='_compute_timesheet_total_duration', help="Total recorded duration, expressed in the encoding UoM, and rounded to the unit")
+    timesheet_total_duration = fields.Integer("Timesheet Total Duration",
+        compute='_compute_timesheet_data', compute_sudo=True,
+        help="Total recorded duration, expressed in the encoding UoM, and rounded to the unit")
```

Comment:
The reason why we use `timesheet_count` rather than `timesheet_total_duration` to determine e.g. if a field is invisible is to improve performance. Putting them back together in the same compute undoes it :(
Is there something I'm overlooking?

Reply:
Oh I see sorry! Reverted this change. Thanks